lib/rules/time-min-milliseconds/index.js

```diff
@@ -52,57 +53,45 @@ const rule = (primary, secondaryOptions) => {
 		root.walkDecls((decl) => {
 			const propertyName = vendor.unprefixed(decl.prop.toLowerCase());
 			const propertyValue = decl.value;
+			const parsedValue = valueParser(getDeclarationValue(decl));
+			let timeValueCount = 0;
+
+			parsedValue.walk((node) => {
+				const { value, sourceIndex } = node;
+				const dimension = getDimension(node);
+
+				if (
+					longhandTimeProperties.has(propertyName) &&
+					!isIgnoredProperty(propertyName) &&
+					!isAcceptableTime(dimension)
+				) {
+					complain(decl, 0, propertyValue.length);
+				}
 
-			if (
-				longhandTimeProperties.has(propertyName) &&
-				!isIgnoredProperty(propertyName) &&
-				!isAcceptableTime(propertyValue)
-			) {
-				complain(decl, 0, propertyValue.length);
-			}
+				if (!shorthandTimeProperties.has(propertyName)) return;
 
-			if (shorthandTimeProperties.has(propertyName)) {
-				const valueListList = postcss.list.comma(propertyValue);
-
-				for (const valueListString of valueListList) {
-					const valueList = postcss.list.space(valueListString);
-
-					if (ignoreDelay) {
-						// Check only duration time values
-						const duration = getDuration(valueList);
-
-						if (duration && !isAcceptableTime(duration)) {
-							complain(decl, propertyValue.indexOf(duration), duration.length);
-						}
-					} else {
-						// Check all time values
-						for (const value of valueList) {
-							if (!isAcceptableTime(value)) {
-								complain(decl, propertyValue.indexOf(value), value.length);
-							}
-						}
-					}
-				}
-			}
+				timeValueCount = calcTimeValueCount(dimension, value, timeValueCount);
+
+				if (isAcceptableTime(dimension) || (ignoreDelay && timeValueCount !== 1)) return;
+
+				complain(decl, sourceIndex, value.length);
+			});
 		});
 
 		/**
-		 * Get the duration within an `animation` or `transition` shorthand property value.
-		 *
-		 * @param {string[]} valueList
-		 * @returns {string | undefined}
+		 * @param {import('postcss-value-parser').Dimension | {unit: null, number: null}} dimension
```

Comment:
[trivial] It seems we can simplify the type a bit as follows, but acceptable to keep it as-is. Perhaps, it's good to change the return type of `getDimension()`.

```suggestion
		 * @param {{unit: string | null, number: string | null}} dimension
```